WIP: GumTree simple

jrfaller opened the pull request:
feat: update to latest gumtree-spoon, update to spoon gumtree-spoon version and replaced matcher with gumtree-simple.

jrfaller commented:
Normally it compiles, however, I cannot run the tests but I am not sure why since it refers to uncompiled file errors, so I suspect this is a build environment or machine configuration problem 😢

monperrus commented:
Revived CI, merged with latest master.

Seems that tests are failing because "because matcher is null"

jrfaller commented:
Thanks! OK, I think I fixed my classpath error, turns out IntelliJ was not setting the right source folders. I made a direct instantiation of the SimpleMatcher and fixed a bug when checking for root nodes. I think it's starting to work I had much more green tests! However, failures are very hard to debug...

jrfaller commented:
OK I have done more experiments.

By reverting to GumTree v4 classic and xy matches, the following test cases do not work:
- mergeToTree_shouldReturnExpectedTree_whenRightVersionIsModified (append_annotation_value_pair)
- mergeToTree_shouldReturnExpectedTree_whenLeftVersionIsModified (append_annotation_value_pair)
- mergeToTree_shouldReturnExpectedTree_whenBothVersionsAreModified (add_similar_methods)
- mergeToTree_shouldReturnExpectedTree_whenBothVersionsAreModified (simple_root_conflict)

Using GumTree v4 simple id matcher
- mergeToTree_shouldReturnExpectedTree_whenRightVersionIsModified (append_annotation_value_pair)
- mergeToTree_shouldReturnExpectedTree_whenLeftVersionIsModified (append_annotation_value_pair)
- mergeToTree_shouldReturnExpectedTree_whenBothVersionsAreModified (add_similar_methods)
- mergeToTree_shouldReturnExpectedTree_whenBothVersionsAreModified (multiple_method_ordering_conflicts)

Therefore it's pretty similar apart from the new failure on multiple_method_ordering_conflicts. I will try to run the code with GumTree v3 classic and xy matchers.

slarse commented:
@jrfaller I'm pretty unavailable this week and next, but I may be able to have a look Sunday (essentially the only day every week where I have some time left over :s).

Something to keep in mind is that some test cases expect _very_ specific behavior. That they fail may not mean that the new result is wrong by definition, it may just be ever so slightly different.

monperrus commented:
> For the annotation value pairs there is definitely a glitch since the key (and not the value) is lost in the merging process. 

Related to https://github.com/SpoonLabs/gumtree-spoon-ast-diff/pull/221 (should be in 1.87 used in this PR)
